shell/platform/linux/public/flutter_linux/fl_value.h

```diff
@@ -0,0 +1,492 @@
+ */
+FlValue* fl_value_map_new();
+
+/**
+ * fl_value_ref:
+ * @value: an #FlValue
+ *
+ * Increase the reference count of an #FlValue.
+ *
+ * Returns: the value that was referenced.
+ */
+FlValue* fl_value_ref(FlValue* value);
+
+/**
+ * fl_value_ref:
+ * @value: an #FlValue
+ *
+ * Derease the reference count of an #FlValue. When the refernece count hits
+ * zero the value is destroyed and no longer valid.
+ */
+void fl_value_unref(FlValue* value);
+
+/**
+ * fl_value_get_type:
+ * @value: an #FlValue
+ *
+ * Get the type of this value.
+ *
+ * Returns: an #FlValueType.
+ */
+FlValueType fl_value_get_type(FlValue* value);
+
+/**
+ * fl_value_equal:
+ * @a: an #FlValue
+ * @b: an #FlValue
+ *
+ * Compare two #FlValue to see if they are equivalent. Two values are considered
+ * equivalent if they are of the same type and their data is the same including
+ * any child values.
+ *
+ * Returns: %TRUE if both values are equivalent.
+ */
+bool fl_value_equal(FlValue* a, FlValue* b);
+
+/**
+ * fl_value_list_add:
+ * @value: an #FlValue of type #FL_VALUE_TYPE_LIST
+ * @child: an #FlValue
+ *
+ * Add a value to a list. Calling this with an #FlValue that is not of type
+ * #FL_VALUE_TYPE_LIST is a programming error.
+ */
+void fl_value_list_add(FlValue* value, FlValue* child);
+
+/**
+ * fl_value_list_add:
+ * @value: an #FlValue of type #FL_VALUE_TYPE_LIST
+ * @child: (transfer full): an #FlValue
+ *
+ * Add a value to a list taking ownership of that value. Calling this with an
+ * #FlValue that is not of type #FL_VALUE_TYPE_LIST is a programming error.
+ */
+void fl_value_list_add_take(FlValue* value, FlValue* child);
+
+/**
+ * fl_value_map_set:
+ * @value: an #FlValue of type #FL_VALUE_TYPE_MAP
+ * @key: an #FlValue
+ * @child_value: an #FlValue
+ *
+ * Set a value in the map. If an existing value was in the map with the same key
+ * it is replaced. Calling this with an #FlValue that is not of type
+ * #FL_VALUE_TYPE_MAP is a programming error.
+ */
+void fl_value_map_set(FlValue* value, FlValue* key, FlValue* child_value);
+
+/**
+ * fl_value_map_set_take:
+ * @value: an #FlValue of type #FL_VALUE_TYPE_MAP
+ * @key: (transfer full): an #FlValue
+ * @child_value: (transfer full): an #FlValue
+ *
+ * Set a value in the map, taking ownership of the keys and values. If an
+ * existing value was in the map with the same key it is replaced. Calling this
+ * with an #FlValue that is not of type #FL_VALUE_TYPE_MAP is a programming
+ * error.
+ */
+void fl_value_map_set_take(FlValue* value, FlValue* key, FlValue* child_value);
+
+/**
+ * fl_value_map_set_string:
+ * @value: an #FlValue of type #FL_VALUE_TYPE_MAP
+ * @key: a UTF-8 text key
+ * @child_value: an #FlValue
+ *
+ * Set a value in the map with a text key. If an existing value was in the map
+ * with the same key it is replaced. Calling this with an #FlValue that is not
+ * of type #FL_VALUE_TYPE_MAP is a programming error.
+ */
+void fl_value_map_set_string(FlValue* value,
```

Comment:
I added this because I expect most of the Map usage is for JSON like objects, i.e.
```c
g_autoptr(FlValue) object = fl_value_map_new ();
fl_value_map_set_string_take (object, "name", name);
fl_value_map_set_string_take (object, "value", fl_value_new_int (42));
```

It's quote common to have these sorts of specializations in GObject APIs, because it would otherwise be unwieldy in C. I certainly appreciate them as an API user.